packages/hardhat-core/src/internal/cli/cli.ts

```diff
@@ -226,6 +233,24 @@ async function main() {
     } else {
       abortAnalytics();
     }
+
+    // VSCode extension prompt for installation
+    const alreadyPrompted = hasPromptedForExtension();
+    if (!alreadyPrompted) {
+      const installationConsent = await confirmExtensionInstallation();
+
+      if (installationConsent !== undefined) {
+        if (installationConsent) {
+          execSync("code --install-extension NomicFoundation.hardhat-solidity");
+        } else {
+          console.log(
+            "To learn more about Hardhat for Visual Studio Code, go to https://hardhat.org/hardhat-vscode"
```

Comment:
@alcuadrado do we want to have this banner even after they've denied consent to install? Guessing yes but wanted to have you confirm.

Reply:
he left a comment on the linear issue mentioning that this is the exact scenario when it should be shown